src/plugins/intel_cpu/src/nodes/fullyconnected.cpp

```diff
@@ -115,7 +115,7 @@ bool FullyConnected::isSupportedOperation(const std::shared_ptr<const ov::Node>&
             errorMessage = "Doesn't support 'data' input with rank: " + std::to_string(inRank);
             return false;
         }
-        if ((one_of(inRank, 2u, 3u) && weightRank != 2) || (inRank == 4 && weightRank != 4)) {
+        if ((one_of(inRank, 2u, 3u) && weightRank != 2) || (inRank == 4 && (!one_of(weightRank, 2u, 4u)))) {
```

Comment:
As far as I know, weights input has always rank == 2: we always reshape the weights to 2D in [ConvertMatMulToFC](https://github.com/openvinotoolkit/openvino/blob/master/src/plugins/intel_cpu/src/transformations/cpu_opset/common/pass/convert_matmul_to_fc.cpp#L142)
```suggestion
        if (one_of(inRank, 2u, 3u, 4u) && weightRank != 2) {
```

Reply:
The weights shape is not always 2D, in [reshape_fc_fusion](https://github.com/openvinotoolkit/openvino/blob/master/src/plugins/intel_cpu/src/transformations/cpu_opset/common/pass/reshape_fc_fusion.cpp#L52-L53) the weight is changed to 4D. For example, input(2,2,3,4), weight(5, 24), output(2,5), weight will be (5,2,3,4) after this transformation.

Reply:
Actually we have PR #21442 which removes the mentioned transformation. Since after this PR we will be able to remove specific handling of 4D input + 4D weights + 2D output from CPU node logic, maybe we can pull changes from #21442 to this PR?

Reply:
Cherry-picked the changes from[ #21442](https://github.com/openvinotoolkit/openvino/pull/21442) and removed the specific handling of 4D input+4D weight+2D output, please review, thanks! @v-Golubev